src/main/java/soonwee/duke/Duke.java

```diff
@@ -0,0 +1,160 @@
+     * done. Macro will end when user types in 'bye'.
+     *
+     * @return updated Task List.
+     * @throws IOException if file or directory is not found.
+     */
+    public static TaskList performSecondTask() throws IOException {
+        String cmd = new String();
+        Storage storage = new Storage("C:\\Users\\songs\\Desktop\\CS2103\\data\\tasks.txt");
+        BufferedReader input = new
+                BufferedReader(new InputStreamReader(System.in));
+        while (true) {
+            cmd = input.readLine();
+            System.out.println(display.printLine());
+            if (cmd.equals("list")) {
+                storage.taskList.displayTasks();
+            } else if (cmd.equals("bye")) {
+                break;
+            } else if (cmd.contains("done")) {
+                int index = Integer.parseInt(cmd.substring(5));
+                storage.taskList.setTaskDone(index);
+                storage.writeFile();
+            } else if (cmd.contains("delete")) {
+                int index = Integer.parseInt(cmd.substring(7));
+                storage.taskList.removeTask(index);
+                storage.writeFile();
+            } else if (cmd.contains("find")) {
+                String text = new Parser().checkFrontInput(cmd, FIND);
+                storage.taskList.searchText(text);
+            } else {
+                storage.taskList = performChildTask(storage.taskList, cmd);
+                storage.writeFile();
+            }
+            System.out.println(display.printLine());
+        }
+        input.close();
+        return storage.taskList;
+    }
+
+    /**
+     * Performs the second macro child task which can add, delete or mark a
+     * task as done. Macro will end when user types in 'bye'.
+     *
+     * @return updated Task List.
+     */
+    public static TaskList performChildTask(TaskList taskList, String cmd) {
+        Parser checker = new Parser();
+        int taskType = checker.checkTaskType(cmd);
+        String task = checker.checkFrontInput(cmd, taskType);
+        if (!task.isEmpty()) {
+            try {
+                if (taskType == -1) {
+                    throw new
+                            DukeException("☹ OOPS!!! I'm sorry, " + "but I don't know what that means :-(");
+                } else {
+                    if (taskType == TODO) {
+                        ToDo newToDo = new ToDo(task);
+                        performAddTask(taskList, newToDo);
+                    } else {
+                        LocalDateTime date = checker.dateFormatter(cmd);
+                        if (taskType == DEADLINE) {
+                            Deadline newDeadLine = new Deadline(task, date);
+                            performAddTask(taskList, newDeadLine);
+                        } else {
+                            Event newEvent = new Event(task, date);
+                            performAddTask(taskList, newEvent);
+                        }
```

Comment:
This looks a little like arrowhead style code. Perhaps you can consider simplifying it so that it's less complicated and indented.

Reply:
Maybe it's better to check the taskType, for example:
`if (taskType == TODO) { 
.....} else if (taskType == DEADLINE) {
.....} else if (taskType == EVENT) {
.....} else {
throw new DukeException(....);
}
`